HabitTracker.Rutwik2005/HabitTracker.Rutwik2005/Program.cs

```diff
@@ -0,0 +1,230 @@
+                            Quantity = reader.GetInt32(2)
+                        }); 
+                }
+            }
+            else
+            {
+                Console.WriteLine("No rows found");
+            }
+            connection.Close();
+            Console.WriteLine("------------------------------\n");
+            foreach (var dw in tableData)
+            {
+                Console.WriteLine($"{dw.Id} - {dw.Date.ToString("dd-MM-yyyy")} - Quantity: {dw.Quantity}");
+            }
+            Console.WriteLine("------------------------------\n");
+        }
+    }
+    private static void Delete()
+    {
+        Console.Clear();
+        GetAllRecords();
+        var recordId = GetNumberInput("\n\nPlease type the ID of the record you want to delete or type 0 to go back to Mian menu\n\n");
+        using (var connection = new SqliteConnection(connectionString))
+        {
+            connection.Open();
+            var tableCmd = connection.CreateCommand();
+            tableCmd.CommandText = $"DELETE from drinking_water WHERE Id = '{recordId}'";
+
+            int rowCount = tableCmd.ExecuteNonQuery();
+
+            if (rowCount == 0)
+            {
+                Console.WriteLine($"\n\nRecord with Id {recordId} does not exist\n\n");
+                Delete();
+            }
+
+        }
+        Console.WriteLine($"\n\nRecord with Id {recordId} was deleted. \n\n");
+        GetUserInput();
+    }
+    internal static void Update()
+    {
+        GetAllRecords();
+
+        var recordId = GetNumberInput("\n\nPlease type Id of the record would like to update. Type 0 to return to main menu.\n\n");
+
+        using (var connection = new SqliteConnection(connectionString))
+        {
+            connection.Open();
+
+            var checkCmd = connection.CreateCommand();
+            checkCmd.CommandText = "SELECT EXISTS(SELECT 1 FROM drinking_water WHERE Id = @id)";
+            checkCmd.Parameters.AddWithValue("@id", recordId);
+            int checkQuery = Convert.ToInt32(checkCmd.ExecuteScalar());
+
+            if (checkQuery == 0)
+            {
+                Console.WriteLine($"\n\nRecord with Id {recordId} doesn't exist.\n\n");
+                connection.Close();
+                Update();
+            }
+
+            string date = GetDateInput();
+            int quantity = GetNumberInput("\n\nPlease insert number of glasses or other measure of your choice (no decimals allowed)\n\n");
+
+            var tableCmd = connection.CreateCommand();
+            tableCmd.CommandText = "UPDATE drinking_water SET date = @date, quantity = @quantity WHERE Id = @id";
+            tableCmd.Parameters.AddWithValue("@date", date);
+            tableCmd.Parameters.AddWithValue("@quantity", quantity);
+            tableCmd.Parameters.AddWithValue("@id", recordId);
+
+            tableCmd.ExecuteNonQuery();
+
+            connection.Close();
+        }
+    }
+
+}
+public class DrinkingWater
```

Comment:
I know the project allows this. But this could easily be another file in your project. Create a DrinkingWater.cs file, cut and paste this in. Bang. More readable, maintainable code and the universe is a little better 😄